### ISSUE TITLE
[RFE] An analysis issue triggered by an indirect dependency looks like a false positive

### ISSUE THREAD

**Issue by mguetta1** [konveyor/analyzer-lsp#882]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Konveyor version

main

### Priority

Undefined (Default)

### Current Behavior

When analyzing the coolstore app with `jakarta-ee` as a target - the following incident looks like a false positive:
Description: Move to Jakarta EE Maven Artifacts - replace groupId javax.activation
RuleID: maven-javax-to-jakarta-00002
Code line:
```
<groupId>javax</groupId>
```

### Expected Behavior

It would be great to be informed somehow that this incident was triggered by an indirect dependency

### How Reproducible

Always (Default)

### Steps To Reproduce

1. Start analysis using Git repository: https://github.com/konveyor-ecosystem/coolstore
2. Analysis mode: Source code + dependencies
3. Set targets: Jakarta EE 9
4. Submit analysis

### Environment

```markdown

```

**Comment by shawn-hurley** [comment]
@mguetta1 is there any way to get the full output of the incident to understand better what is happening?

**Comment by shawn-hurley** [comment]
@mguetta1 Any chance we can get an example of this?

**Comment by shawn-hurley** [comment]
Here is the snippet to make sure that we are talking about the same thing:

```yaml
- id: 2
  createTime: 2024-10-14T18:45:42.349806325Z
  analysis: 1
  ruleset: eap7/weblogic/tests/data
  rule: maven-javax-to-jakarta-00002
  name: ""
  description: Move to Jakarta EE Maven Artifacts - replace groupId javax.activation
  category: potential
  effort: 1
  incidents:
  - id: 2
    createTime: 2024-10-14T18:45:42.349902566Z
    issue: 2
    file: /shared/source/coolstore/pom.xml
    line: 25
    message: If you migrate your application to JBoss EAP 7.3, or later, and want
      to ensure its Maven building, running or testing works as expected, use instead
      the Jakarta EE dependency with groupId `com.sun.activation`
    codeSnip: |-
      16      </properties>
      17      <dependencies>
      18          <dependency>
      19              <groupId>javax</groupId>
      20              <artifactId>javaee-web-api</artifactId>
      21              <version>7.0</version>
      22              <scope>provided</scope>
      23          </dependency>
      24          <dependency>
      25              <groupId>javax</groupId>
      26              <artifactId>javaee-api</artifactId>
      27              <version>7.0</version>
      28              <scope>provided</scope>
      29          </dependency>
      30          <dependency>
      31              <groupId>org.jboss.spec.javax.jms</groupId>
      32              <artifactId>jboss-jms-api_2.0_spec</artifactId>
      33              <version>2.0.0.Final</version>
      34          </dependency>
      35          <dependency>
      36              <groupId>org.flywaydb</groupId>
    facts:
      name: javax.activation.activation
      version: "1.1"
  links:
  - url: https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.3/html-single/migration_guide/index#maven-artifact-changes-for-jakarta-ee_default
    title: 'Red Hat JBoss EAP 7.3 Migration Guide: Maven Artifact Changes for Jakarta
      EE'
  labels:
  - JakartaEE
  - konveyor.io/source
  - konveyor.io/target=eap
  - konveyor.io/target=eap7
  - konveyor.io/target=eap8
  - konveyor.io/target=jakarta-ee
  - konveyor.io/target=jakarta-ee8
 ```

**Comment by jmle** [comment]
Did we create the RFE at the end? @rromannissen fyi, although I think we might have already talked about this

**Comment by rromannissen** [comment]
Isn't this a false positive though? Does it make sense to flag transitive dependencies when we don't know if they are actually being used in the application itself? I guess the actual question is, should the java.dependency lookup include transitive dependencies? Is this really valuable information for the user when we can't confirm if the transitive dependencies are actually being used in the application itself? 

In this specific case, we are talking about a ruleset that covers the EAP 7 to 8 migration path. The issue was detected on the javax:javaee-api:7.0 dependency for this application. javax.activation:activation:1.1 is not even a direct dependency of javax:javaee-api:7.0, but a transitive dependency for that library via the com.sun.mail:javax.mail:1.5.0 direct dependency, which is in turn a transitive dependency for the application. Right now, we are flagging the javax:javaee-api:7.0 dependency and providing information about a transitive dependency that is two levels below in the dependency tree. That doesn't seem to be useful information for the end user reading the report IMHO. Considering the migration path we're in, I think the useful information for the user would be to flag javax:javaee-api:7.0 by itself and offer the equivalent Jakarta EE artifact.

**Comment by jmle** [comment]
I think there are several things we could do here:

1. Do not match indirect dependencies in dependency rules.
2. Match them, but flag them somehow in the output.
3. Add a field in dependency rules that indicates whether the rule should match also indirect dependencies or not and act accordingly.

(3) might be the best, since this whole thing might depend on the rule itself. If we go this way we need to choose a default for this new field, so that we don't need to change all rules. WDYT?

**Comment by rromannissen** [comment]
Some field called "matchWhenTransitive" or something similar that defaults to false sounds like a great idea! There might be scenarios in which a rules author might not want a dependency to be included even when it is transitive, so that makes sense. Nevertheless, 3 kind of requires 2 to be practical. If the match is produced via the field being true, then we need a way to flag these transitive dependencies for the output to make sense. We need to consider that the rules author and the rules consumer are different personas, and the consumer is the one that needs to understand the analysis output.
